src/Corax/Mappings/IndexFieldsMapping.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Sparrow.Server;
+using Voron;
+
+namespace Corax.Mappings;
+
+public class IndexFieldsMapping : IEnumerable<IndexFieldBinding>, IDisposable
+{
+    private readonly Dictionary<Slice, IndexFieldBinding> _fields;
+    private readonly Dictionary<int, IndexFieldBinding> _fieldsById;
+    private readonly ByteStringContext _mappingContext;
+    public readonly Func<string, Analyzer> ExactAnalyzer;
+    public readonly Func<string, Analyzer> SearchAnalyzer;
+    public readonly Analyzer DefaultAnalyzer;
+    public readonly int MaximumOutputSize;
+    public readonly int MaximumTokenSize;
+    public int Count => _fields.Count;
+
+    internal IndexFieldsMapping(ByteStringContext mappingContext, Dictionary<Slice, IndexFieldBinding> fields, Dictionary<int, IndexFieldBinding> fieldsById, Analyzer defaultAnalyzer, Func<string, Analyzer> searchAnalyzer, Func<string, Analyzer> exactAnalyzer, int maximumOutputSize, int maximumTokenSize)
+    {
+        _mappingContext = mappingContext;
```

Comment:
Who owns this? How can we be sure that if we are passed this context, it isn't disposed before the mapping?

Reply:
The child of a builder, in this case, the owner is `IndexFieldsMapping`. Unless there is an exception during building, then the owner of the context is a builder. I've changed the builder to be `IDisposable`.